dev/com.ibm.jbatch.container/src/com/ibm/jbatch/container/services/impl/JPAPersistenceManagerImpl.java

```diff
@@ -2712,7 +2717,8 @@ public Integer call() throws Exception {
             }.runInNewOrExistingGlobalTran();
 
             return exec;
-        } catch (javax.persistence.PersistenceException pe) {
+        } catch (PersistenceException pe) {
+            logger.fine("Caught a persistence exception");
             Throwable cause = pe.getCause();
             while (cause != null) {
```

Comment:
Let's trace each cause along the chain, printing the exc itself and the exc.getMessage either here or at the end of the loop if we haven't returned.     THis way if we step through a chain and don't match we have evidence along the way